nucypher/utilities/prometheus/metrics.py

```diff
@@ -0,0 +1,313 @@
+        request.setHeader(b'Content-Type', "text/json")
+        return self.generate_latest_json()
+
+    @staticmethod
+    def get_exemplar(sample, metric):
+        if not sample.exemplar:
+            return {}
+        elif metric.type not in ('histogram', 'gaugehistogram') or not sample.name.endswith('_bucket'):
+            raise ValueError(
+                "Metric {} has exemplars, but is not a "
+                "histogram bucket".format(metric.name)
+            )
+        return {
+            "labels": sample.exemplar.labels,
+            "value": floatToGoString(sample.exemplar.value),
+            "timestamp": sample.exemplar.timestamp
+        }
+
+    def get_sample(self, sample, metric):
+        return {
+            "sample_name": sample.name,
+            "labels": sample.labels,
+            "value": floatToGoString(sample.value),
+            "timestamp": sample.timestamp,
+            "exemplar": self.get_exemplar(sample, metric)
+        }
+
+    def get_metric(self, metric):
+        return {
+            "samples": [self.get_sample(sample, metric) for sample in metric.samples],
+            "help": metric.documentation,
+            "type": metric.type
+        }
+
+    def generate_latest_json(self):
+        """
+        Returns the prometheus from the registry
+        in latest JSON format as a string.
+        """
+        output = {}
+        for metric in self.registry.collect():
+            try:
+                output[metric.name] = self.get_metric(metric)
+            except Exception as exception:
+                exception.args = (exception.args or ('',)) + (metric,)
+                raise
+
+        json_dump = json.dumps(output, cls=MetricsEncoder).encode('utf-8')
+        return json_dump
+
+
+def collect_prometheus_metrics(metrics_collectors: List[MetricsCollector], node_metrics: Dict) -> None:
+    for collector in metrics_collectors:
+        collector.collect(node_metrics)
+
+
+def initialize_prometheus_exporter(ursula, prometheus_config: PrometheusMetricsConfig) -> None:
+    from prometheus_client.twisted import MetricsResource
+    from twisted.web.resource import Resource
+    from twisted.web.server import Site
+
+    metrics_prefix = prometheus_config.metrics_prefix
+    metrics_collectors = create_metrics_collectors(ursula, metrics_prefix)
+    # initalized collectors
+    node_metrics = {}
+    for collector in metrics_collectors:
+        node_metrics.update(collector.initialize(metrics_prefix=metrics_prefix, registry=ursula.registry))
+
+    # TODO: was never used
+    # "requests_counter": Counter(f'{metrics_prefix}_http_failures', 'HTTP Failures', ['method', 'endpoint']),
+
+    # Scheduling
+    metrics_task = task.LoopingCall(collect_prometheus_metrics,
+                                    metrics_collectors=metrics_collectors,
+                                    node_metrics=node_metrics)
+    metrics_task.start(interval=10, now=False)  # TODO: make configurable
```

Comment:
Heh, might need a running list of configurations. It does not need to anything near comprehensive, but perhaps some high-level tuning knobs.